README.md

```diff
@@ -1,46 +1,70 @@
 # Lindos Tray App
 
-Your friendly desktop assistent. A tray icon for tedious chores and lovely conversations.
+Your friendly desktop assistant. A tray icon for tedious chores and lovely conversations.
 
 ## Quick Start
 
-### Prerequisites
+This project uses [Just](https://just.systems/) for common development tasks. Install it first:
+
+```bash
+# macOS
+brew install just
 
-- macOS with Xcode (including the command-line tools) for the tray app build.
-- Rust toolchain with `cargo` (install via [rustup](https://rustup.rs/)).
+# Linux
+cargo install just
+```
+
+Then see all available commands:
 
 ```bash
-# build Rust + macOS app
-scripts/build_macos.sh debug
+just --list
 ```
 
-Or open `macos/LindosTrayApp/LindosTrayApp.xcodeproj` and hit Run in Xcode.
+### Prerequisites
 
-## Run Rust Tests
+Install development prerequisites for your platform:
 
 ```bash
-# from the repo root
-cd rust-core
-cargo test
+# Rust (macOS and Linux)
+just install-rust
+
+# macOS with Xcode
+just install-swift-macos
+
+# GNOME development on NixOS
+just install-gnome-nixos
+
+# Python tools (uv)
+just install-python-tools
+
+# Code coverage tool
+just install-coverage-tool
 ```
 
-## Code Coverage
+### Build & Run
+
+```bash
+# Build macOS app
+just build-macos-debug
+
+# Or open in Xcode
+open macos/LindosTrayApp/LindosTrayApp.xcodeproj
+```
 
-Generate test coverage reports locally using [cargo-llvm-cov](https://github.com/taiki-e/cargo-llvm-cov):
+### Development
 
 ```bash
-# Install cargo-llvm-cov (one-time setup)
-cargo install cargo-llvm-cov
+# Run all Rust checks (clippy, formatting, tests)
+just check-rust
 
-# Generate coverage report in terminal
-cd rust-core
-cargo llvm-cov
+# Run all Python checks (ruff, pytest)
+just check-python
 
-# Generate HTML coverage report (opens in browser)
-cargo llvm-cov --open
+# Generate code coverage report
+just coverage
 
-# Generate LCOV format for integration with other tools
-cargo llvm-cov --lcov --output-path lcov.info
+# Generate and open HTML coverage report
+just coverage-html
 ```
 
-Coverage is automatically measured in CI and reported to Codecov on pull requests.
+See `just --list` for all available commands.
```

Comment:
This list will easily outdate. Is there a convincing benefit of keeping this list of commands when we get everything listed simply by running `just`?